python/ray/air/util/tensor_extensions/arrow.py

```diff
@@ -321,6 +321,8 @@ def from_numpy(
                 # NOTE: Arrow expects LSB bit-packed ordering.
                 # NOTE: This creates a copy.
                 arr = np.packbits(arr, bitorder="little")
+            if pa.types.is_string(pa_dtype):
```

Comment:
This adds support for homogeneous-shaped tensor columns; it seems like this support might easily extend to our variable-shaped tensor column type. If adding support is straightforward, could we add it in this PR? https://github.com/ray-project/ray/blob/91d86fe72e0098d38dbac67a59b7b1f1ebe12984/python/ray/air/util/tensor_extensions/arrow.py#L622

Support for ragged strings tensors would be great for a few NLP use cases, e.g. [textual sentences/documents of different lengths](https://towardsdatascience.com/using-tensorflow-ragged-tensors-2af07849a7bd), and [N-grams](https://www.tensorflow.org/guide/ragged_tensor#example_use_case).